TODO.md

```diff
@@ -0,0 +1,8 @@
+- [x] SDL should extend type for external types - I guess marking types in SDL
+  - [ ] can't generate graphql-js stuff, don't want to do it for externs - don't support graphql-js for this?
+- [ ] all imported types (so support interfaces etc)
+- [ ] Read SDL to actually do validation
+  - [ ] reenable global validations
+- [ ] "modular" mode? like no full schema, but parts of schema but with full validation by resolving it?
+- [ ] all tests to add fixtures for metadata/resolver map
+- [ ] pluggable module resolution - too many variables there, use filepath by default, let users customize it
```

Comment:
In theory we only need this path in order to perform validation (is that right?). I wonder if there is a TypeScript API which will let us use the same (or most of) TypeScript's implementation of module resolution. I would need to avoid expecting the file to have a .js/.ts/.mjs/etc extension, but maybe something like that is exposed? I suspect that would be sufficient we could find something like that.

I did see this: https://github.com/microsoft/TypeScript/blob/6a00bd2422ffa46c13ac8ff81d7b4b1157e60ba7/src/server/project.ts#L484

Which could be a good place to start looking.

Reply:
I'm not sure any non-ts files are included in how the typescript resolves it. There is also the whole story of "lib" in package.json and stuff like this. Ultimately there is no "well-known" spot for the GraphQL files, so it feels like it all might just break weirdly. I will experiment on this ofc, but I feel that there might not be an easy solution.